### PR TITLE
allow settle fees to silently fail

### PR DESCRIPTION
**Pull request by mschneider**
this is supposed to enable closing out a whole account in a single tx

### CODE REVIEW

**Review by microwavedcola1 — APPROVED** [review]

File: program/src/processor.rs [hunk 1]
```diff
@@ -3083,8 +3083,10 @@ impl Processor {
         let contract_size = mango_group.perp_markets[market_index].base_lot_size;
         let new_quote_pos = I80F48::from_num(-pa.base_position * contract_size) * price;
         let pnl: I80F48 = pa.quote_position - new_quote_pos;
-        check!(pnl.is_negative(), MangoErrorCode::Default)?;
-        check!(perp_market.fees_accrued.is_positive(), MangoErrorCode::Default)?;
+        // ignore these cases and fail silently so transactions can continue
+        if pnl.is_negative() || perp_market.fees_accrued.is_positive() {
```
Comment (dafyddd) [inline program/src/processor.rs:3087]: I think you want `if !(pnl.is_negative() && perp_market.fees_accrued.is_positive())`

Comment (mschneider) [reply program/src/processor.rs:3087]: right

Comment (dafyddd) [reply program/src/processor.rs:3087]: I think you should also print a `msg!` explaining why it's failing silently. We've used that standard elsewhere

Comment (microwavedcola1) [reply program/src/processor.rs:3087]: hmm, good catch, embarrassing oversight from me, no change is small ;)

**Pull request merged**